### ISSUE TITLE
[FEATURE] onAnimationComplete pass variant as param

### ISSUE THREAD

**Issue by klase** [framer/motion#316]
Currently I'm not sure how to determine which animation variant completed in the onAnimationComplete callback. For instance I'd like to know when the exit transition has fully completed inside AnimatePresence so I can disable scroll locking after closing a modal. In Pose I used:

```js
onPoseComplete={thePose => {
  if (thePose === "exit") {
    doStuff();
  }
}
```

**Describe the solution you'd like**
Pass the related variant string as argument to onAnimationComplete like onPoseComplete? 

Edit: 
Noticed I can use the onExitComplete callback on AnimatePresence itself but would still be useful to know which animation completed on a motion level.

**Comment by Sandstedt** [comment]
Any progress with this? Or any workarounds?

**Comment by nathanredblur** [comment]
I also searching for a solution, also problems with this approach:

```
export const Alert:React.FC<Props> = ({isVisible}) => {
  console.log(isVisible)
  const onComplete = () => {
    if (isVisible) {
      console.log('afterOpen')
    } else {
      console.log('afterClose')
    }
  }

  return (
    <AnimatePresence>
      {isVisible && (
        <div key="modal">
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className={styles.AlertBackground}
            onAnimationComplete={onComplete}
          />
      )}
    </AnimatePresence>
  )
}
```

I get this output:
```
false
true
afterOpen
false
afterOpen
```

look like `onAnimationComplete` is not updating the function

**Comment by nathanredblur** [comment]
also will be nice something like `<AnimatePresence onAnimationComplete={afterEnter}>`

**Comment by nathanredblur** [comment]
I did this workaround:

```TS
export const Alert:React.FC<Props> = ({isVisible}) => {
    const [isCompleted, setIsCompleted] = useState(false)
  useEffect(()=> {
    if (isCompleted) {
      if (isVisible) {
        console.log('afterOpen')
      } else {
        console.log('afterClose')
      }
      setIsCompleted(false)
    }
  }, [isCompleted, isVisible])

  
  const onComplete = () => {
    setIsCompleted(true)
  }

  return (
    <AnimatePresence>
      {isVisible && (
        <div key="modal">
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className={styles.AlertBackground}
            onAnimationComplete={onComplete}
          />
      )}
    </AnimatePresence>
  )
}
```

**Comment by alex83130** [comment]
Any update on that one ? It's too bad to have to use another external state in order to make the difference between the different state :/

**Comment by mattgperry** [comment]
Published in 3.6
